features/promotion/managing_catalog_promotions/creating_catalog_promotion.feature

```diff
@@ -132,6 +132,23 @@ Feature: Creating a catalog promotion
         And it should have "winter_sale" code and "Winter sale" name
         And it should have priority equal to 10
 
+    @api @ui @javascript
+    Scenario: Creating a new catalog promotion with priorities with other exclusive already existing
```

Comment:
If I see correctly, this scenario does not cover the problem we have, because it should also work before your changes in implementation. You should create a scenario in which you create a catalog promotion with such a priority that the new promotion will be between the existing ones, then we will be able to check if all catalog promotions have been reapplied, not only the new one.

Reply:
and probably this CP should be with a price fixed discount (all percentage discounts in any order will decrease price in the same level :))

Reply:
yeah, true, new stem should cover all cases as its mixed with percentage and fixed discount